src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -245,6 +245,21 @@ public interface ITransformer
         /// Note that <see cref="IDataView"/>'s are lazy, so no actual transformations happen here, just schema validation.
         /// </summary>
         IDataView Transform(IDataView input);
+
+        /// <summary>
+        /// Whether a call to <see cref="GetRowToRowMapper(ISchema)"/> should succeed, on an
+        /// appropriate schema.
+        /// </summary>
+        bool IsRowToRowMapper { get; }
+
+        /// <summary>
+        /// Constructs a row-to-row mapper based on an input schema. If <see cref="IsRowToRowMapper"/>
+        /// is <c>false</c>, then an exception should be thrown. If the input schema is in any way
+        /// unsuitable for constructing the mapper, an exception should likewise be thrown.
```

Comment:
>an exception should likewise be thrown [](start = 52, length = 38)

I think this should be checked in `TestEstimatorCore` somewhere

Reply:
Yeah, you're probably right... let me check the schemas.

---
In reply to: [219580497](https://github.com/dotnet/machinelearning/pull/973#discussion_r219580497) [](ancestors = 219580497)